File: backend/src/main/java/ch/helsanahack/backend/entities/BmiEntry.java
package ch.helsanahack.backend.entities;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.Id;
import javax.persistence.Table;

@Entity
@Data
@Table(name = "weight")
@NoArgsConstructor
@AllArgsConstructor
public class BmiEntry {

    @Id
    @Column(name = "datauuid")
    private String id;

    @Column(name = "weight")
    private Double weight;

    @Column(name = "height")
    private Double height;
}

File: frontend/src/models/BmiEntry.ts
export class BmiEntry {
  id: string;
  weight: number;
  height: number;

  constructor(id: string, weight: number, height: number) {
    this.id = id;
    this.weight = weight;
    this.height = height;
  }

  calculateBmi(): number {
    return (this.weight / ((this.height * this.height / 10000)));
  }
}
